build.gradle

```diff
@@ -53,26 +56,61 @@ dependencies {
     testImplementation 'junit:junit:4.12'
 }
 
-// create a release zip
-task release(type: Zip) {
+def readModInfoJson() {
     def jsonFile = file('src/main/resources/mod_info.json')
-    def slurper = new groovy.json.JsonSlurper()
-    def parsedJson = slurper.parse(jsonFile)
+    def slurper = new JsonSlurper()
+    return slurper.parse(jsonFile)
+}
+
+task release(type: Zip) {
     dependsOn build
-    archiveFileName = "${project.name}-${parsedJson.version}.zip"
+    archiveFileName = "${project.name}-${readModInfoJson().version}.zip"
     destinationDirectory = file(rootDir)
     from sourceSets.main.resources
     from jar
     into project.name
 }
 
+def arePropertiesDefined() {
+    if (!project.hasProperty('game.dir') || !project.hasProperty('game.mod_folder')) {
+        println "WARNING: Missing properties in gradle.properties. See gradle.properties.dist as an example."
+        println "WARNING: Task execution stopped."
+        return false;
+    }
+    return true;
+}
+
+def getModDir() {
+    return project.getProperty('game.dir') + project.getProperty('game.mod_folder') + readModInfoJson().id
+}
+
 task copyToMods(type: Copy) {
-    if (!project.hasProperty('game.full_path')) {
-        println "WARNING: Missing game.full_path in gradle.properties. See gradle.properties.dist as an example."
+    if (!arePropertiesDefined()) {
         return;
     }
 
     dependsOn build
     from jar
-    into project.getProperty('game.full_path')
-}
\ No newline at end of file
+
+    into getModDir()
+}
+
+def getCoreDir() {
+    return project.getProperty('game.dir') + "/starsector-core/"
```

Comment:
Why do you need core dir?

Reply:
That's where the .bat file is for windows and in the .bat file I define extra parameters that are not present when I just run the executable from the game root directory. Bat file parameters are not matching with root dir file "vmparams".

Reply:
That's shit, starsector.sh is in main dir :(

Reply:
E.g. there is no starsector-core, it's all inside:
```
commons-compiler.jar      hs_err_pid316376.log  json.jar         native             starfarer_obf.jar   starsector.log.1
commons-compiler-jdk.jar  janino.jar            LICENSE.txt      saves              starsector.bat      starsector.log.2
data                      jinput.jar            log4j-1.2.9.jar  screenshots        starsector.bat32    starsector.log.3
fs.common_obf.jar         jogg-0.0.7.jar        lwjgl.jar        sounds             starsector.command  starsector.png
fs.sound_obf.jar          jorbis-0.0.15.jar     lwjgl_util.jar   starfarer.api.jar  starsector.icns     starsector.sh
graphics                  jre_linux             mods             starfarer.api.zip  starsector.log      xstream-1.4.10.jar
```